addons/mail/models/mail_message.py

```diff
@@ -81,6 +81,9 @@ def _get_default_author(self):
     needaction = fields.Boolean(
         'Need Action', compute='_get_needaction', search='_search_needaction',
         help='Need Action')
+    haserror = fields.Boolean(
```

Comment:
This is inspired from the needaction. I think that the compute_sudo won't have the same effect since we want to find all notifications but we do not sudo on message. (for message in self and not for messages in self.sudo(). Does it makes sense?